src/main.cpp

```diff
@@ -4667,15 +4696,26 @@ bool LoadExternalBlockFile(const CChainParams& chainparams, FILE* fileIn, CDiskB
 
                 // process in case the block isn't known yet
                 if (mapBlockIndex.count(hash) == 0 || (mapBlockIndex[hash]->nStatus & BLOCK_HAVE_DATA) == 0) {
+                    LOCK(cs_main);
                     CValidationState state;
-                    if (ProcessNewBlock(state, chainparams, NULL, &block, true, dbp, false))
+                    if (AcceptBlock(block, state, chainparams, NULL, true, dbp))
```

Comment:
let's not aggressively change code to c++11 because 1.0 release is not using it.  Its perfectly fine to do it opportunistically if you want, but not mandatory for PR acceptance.